code/game/objects/items/books/skill_book.dm

```diff
@@ -23,28 +23,31 @@
 										"\"[skill_name] For Younglets\"", \
 										"\"[skill_name]: Volume [rand(1,100)]\"", \
 										"\"Understanding [skill_name]: [rand(1,100)]\th Edition\"", \
-										"\"Dealing With Ungrateful Customers Dissatisfied With Your Perfectly Acceptable [skill_name] Services\""))
+										"\"Dealing With Ungrateful Customers Dissatisfied With Your Perfectly Acceptable [skill_name] Services\"", \
+										"\"Really big book of [skill_name]\""))
 
 /*
 Skill books that increase your skills while you activate and hold them
 */
 
 /obj/item/book/skill
-	name = "default textbook" // requires default names for tradershop, cant rely in Initialize for names
+	name = "default textbook" // requires default names for tradershop, cant rely on Initialize for names
 	desc = "A blank textbook. (Notify admin)"
 	author = "The Oracle of Bakersroof"
 	icon_state = "book2"
 	force = 4
-	w_class = ITEM_SIZE_HUGE //Skill books are THICC with knowledge. Just to stop people carrying around a library in a box
+	w_class = ITEM_SIZE_LARGE //Skill books are THICC with knowledge. Up one level from regular books to prevent library-in-a-bag silliness.
 	unique = 1
+	matter = list(MAT_PLASTIC = SHEET_MATERIAL_AMOUNT*1.2, MAT_WOOD = SHEET_MATERIAL_AMOUNT*1.2) //20% more than regular books. They're bigger and more valuable ok?
 
 	var/decl/hierarchy/skill/skill // e.g. SKILL_LITERACY
 	var/skill_req //The level the user needs in the skill to benefit from the book, e.g. SKILL_PROF
 	var/reading = FALSE //Tto check if the book is actively being used
-	var/custom = FALSE //To bypass init stuff, for player made textbooks and weird books
+	var/custom = FALSE //To bypass init stuff, for player made textbooks and weird books. If true must have details manually set
 	var/ez_read = FALSE //Set to TRUE if you can read it without basic literacy skills
 
 	var/skill_name = "missing skill name"
+	var/progress = 6 // used to track the progress of making a custom book. defaults to 6 so most books count as 'finished'
```

Comment:
Can you elaborate or give an example?

Reply:
You'd define something like 
````
#define TEXTBOOK_PROGRESS_NONE 0
...
#define TEXTBOOK_PROGRESS_FINISHED 6
````
then rather than using magic numbers in your code you'd use the defines.

Reply:
Okay, think I got this one.